server/src/internalClusterTest/java/org/opensearch/indices/replication/SegmentReplicationDefaultIT.java

```diff
@@ -0,0 +1,112 @@
+            .put(IndexModule.INDEX_QUERY_CACHE_ENABLED_SETTING.getKey(), false)
+            .build();
+    }
+
+    @Override
+    protected Settings featureFlagSettings() {
+        return Settings.builder().put(super.featureFlagSettings()).put(FeatureFlags.REPLICATION_TYPE, "true").build();
+    }
+
+    @Override
+    protected boolean addMockInternalEngine() {
+        return false;
+    }
+
+    @Override
+    protected Settings nodeSettings(int nodeOrdinal) {
+        return Settings.builder().put(super.nodeSettings(nodeOrdinal)).put(DEFAULT_REPLICATION_TYPE_SETTING_SEGMENT.getKey(), true).build();
+    }
+
+    public static class TestPlugin extends Plugin implements SystemIndexPlugin {
+        @Override
+        public Collection<SystemIndexDescriptor> getSystemIndexDescriptors(Settings settings) {
+            return Collections.singletonList(
+                new SystemIndexDescriptor(SYSTEM_INDEX_NAME, "System index for [" + getTestClass().getName() + ']')
+            );
+        }
+    }
+
+    @Override
+    protected Collection<Class<? extends Plugin>> nodePlugins() {
+        return Arrays.asList(ClusterInfoServiceIT.TestPlugin.class, MockTransportService.TestPlugin.class);
+    }
+
+    public void testReplicationWithDefaultSegmentReplicationSetting() throws Exception {
+
+        boolean isSystemIndex = randomBoolean();
+        String indexName = isSystemIndex ? SYSTEM_INDEX_NAME : INDEX_NAME;
+
+        // Starting two nodes with primary and replica shards respectively.
+        final String primaryNode = internalCluster().startNode();
+        createIndex(indexName);
+        ensureYellowAndNoInitializingShards(indexName);
+        final String replicaNode = internalCluster().startNode();
+        ensureGreen(indexName);
+
+        final int initialDocCount = scaledRandomIntBetween(20, 30);
+        for (int i = 0; i < initialDocCount; i++) {
+            client().prepareIndex(indexName).setId(Integer.toString(i)).setSource("field", "value" + i).execute().actionGet();
+        }
+
+        refresh(indexName);
+        assertBusy(() -> {
+            assertHitCount(client(replicaNode).prepareSearch(indexName).setSize(0).setPreference("_only_local").get(), initialDocCount);
+        });
+
+        SegmentReplicationStatsResponse segmentReplicationStatsResponse = client().admin()
+            .indices()
+            .prepareSegmentReplicationStats(indexName)
+            .execute()
+            .actionGet();
+        if (isSystemIndex) {
+            // Verify that Segment Replication did not happen on the replica shard.
+            assertNull(segmentReplicationStatsResponse.getReplicationStats().get(indexName));
+        } else {
+            // Verify that Segment Replication happened on the replica shard.
+            assertFalse(segmentReplicationStatsResponse.getReplicationStats().get(indexName).get(0).getReplicaStats().isEmpty());
+        }
+    }
```

Comment:
We could add some more tests here to test the behavior described:

1. A test to ensure that when cluster setting for segrep is true, creating an index with docrep enabled overrides this cluster setting.
2. Similarly a test to ensure overriding takes place when index.replication.type is set to docrep when the segrep cluster setting is true. 
3. A last test (optional) to check that when the feature flag is not enabled, even setting the cluster setting doesn't result in segment replication indices.